c/src/messenger/store.c

```diff
@@ -421,7 +421,7 @@ int pni_store_update(pni_store_t *store, pn_sequence_t id, pn_status_t status,
     }
   }
 
-  while (store->hwm - store->lwm > 0 &&
+  while (store->hwm > store->lwm &&
```

Comment:
I'd feel safer replacing this with `store->hwm != store->lwm` just to keep the code behavior completely identical. I haven't managed to convince myself that this change is always going to work... Somebody else who actually understands this needs to review as well.

Reply:
See my comment - the change is incorrect.